src/telegram/ext/_utils/networkloop.py

```diff
@@ -141,13 +141,52 @@ async def do_action() -> None:
             )
             # pylint: disable=protected-access
             cur_interval = slack_time + exc._retry_after.total_seconds()
+
+            # Check max_retries for RetryAfter as well
+            if on_err_cb:
+                on_err_cb(exc)
+
+            if 0 <= max_retries <= retries:
+                _LOGGER.exception(
+                    "%s Failed run number %s of %s. Aborting.", log_prefix, retries, max_retries
+                )
+                raise
```

Comment:
1) we should also have a log issued in case we do retry - see how it's handled in the `TelegramError` clause
2) please avoid duplication of this `if-else` logic, as it's now repeated in `RetryAfter`, `TimedOut` and `TelegramError`. One option would be to intrudce a small local helper function that checks the max-try condition and issues the logs accordingly